pyccel/ast/bitwise_operators.py

```diff
@@ -68,9 +66,8 @@ class PyccelBitOperator(PyccelOperator):
     arg2: PyccelAstNode
         The second argument passed to the operator
     """
-    __slots__ = ()
```

Comment:
All classes should define `__slots__` even if it contains nothing

Reply:
In this class the `__slots__` should be `_precision`. The `_dtype`, `_shape` and `_rank` are all the same for all instances of the class so they should be class properties